group_processor.go

```
@@ -91,6 +91,38 @@ func (gp *GroupProcessor) New() (err error) {
 	return nil
 }
 
+func (gp *GroupProcessor) SetOffset(partition int32, offset int64) error {
+	gp.RLock()
+	defer gp.RUnlock()
+
+	oldest, err := gp.kafka.client.GetOffset(gp.Topic, partition, sarama.OffsetOldest)
+
+	if err != nil {
+		return err
+	}
+
+	newest, err := gp.kafka.client.GetOffset(gp.Topic, partition, sarama.OffsetNewest)
+
+	if err != nil {
+		return err
+	}
+
+	if offset < oldest {
+		gp.log.Warnf("Offset of %s/%d out of range. Corrected offset from %d to %d.", gp.Topic, partition, offset, oldest)
+		offset = oldest
```

This implicit behaviour might be dangerous in certain cases - the client may not want to resume if the offset is out of range. What do you think about introducing an extra boolean argument - something like `correctOffsets`? If it's `false`, we can return an error.